### PR TITLE
:book: Add frontmatter to reference docs and transform attribute listings to tables

### PR DESCRIPTION
**Pull request by matthiasrohmer**
- Adds frontmatter to a lot of the component reference markdowns in the form of
```yaml
$category@: media | ads-analytics | dynamic-content | layout | media | presentation | social
formats:
  - websites | stories | ads | email
teaser:
  text: Displays GL Transmission Format (gITF) 3D models.
```
This makes reusage and importing of the docs for ampproject.org/amp.dev a lot easier.

- Transforms a lot of the attribute sections from lists to tables

/cc @pbakaus

### CODE REVIEW

**Comment by pbakaus** [comment]
@cramforce this would make it much simpler to maintain amp.dev ref docs, at the expense of making it a little uglier to look at these docs when browsing in this repo. LMK if you have any concerns or questions. This is a big diff, but the content hasn't been touched.

**Comment by pbakaus** [comment]
@matthiasrohmer what are the CLA issues here? Same as before? Want to make sure this isn't a blocker for merging.

**Comment by matthiasrohmer** [comment]
> @matthiasrohmer what are the CLA issues here? Same as before? Want to make sure this isn't a blocker for merging.

Just saw it 😅 At least the signing of the company CLA went fine this time. I just need everyone to join the corresponding Google Group then it should work. But if you want to force merge to get things done that's also fine for me.

**Comment by pbakaus** [comment]
@matthiasrohmer please definitely have people join the google group. I'll let the current merge who sees this PR decide, but should be safe.

**Comment by matthiasrohmer** [comment]
So all colleagues who contributed to this PR have joined the Google Group jungvonmatt-tech for which there should be a valid company CLA. Also they were using the e-mail addresses that they've used to push and that one is also linked to their GitHub profile. Theoretically we are set. @googlebot, WDYT?

**Comment by cramforce** [comment]
To simplify the merge (and reasoning about the CLA), please squash to one commit.

**Comment by pbakaus** [comment]
@matthiasrohmer the Travis CI here (see https://travis-ci.org/ampproject/amphtml/jobs/490246695) is failing as you touched some files that had bad links in them. Could you take a look and fix the whitelist as documented in the error thrown?

File: extensions/amp-gfycat/amp-gfycat.md [hunk 1]
```diff
@@ -70,7 +76,7 @@ Gfycat allows many aspect ratios.
 
 To specify the width and height in the code, copy it from the embed URL:
 
-1. Go to https://gfycat.com/name, where name is the Gfycat ID.
+1. Go to [https://gfycat.com/](https://gfycat.com/)`name`, where name is the Gfycat ID.
```
Comment (rsimha) [inline extensions/amp-gfycat/amp-gfycat.md]: From looking at https://travis-ci.org/ampproject/amphtml/jobs/490580706#L810, this is being treated as a single link. Perhaps you can put the entire string in backticks, like this?
```
Go to `https://gfycat.com/name`, where name is the Gfycat ID.
```

Comment (matthiasrohmer) [reply extensions/amp-gfycat/amp-gfycat.md]: Saw your comment to late, I went with the following. What do you think? 

```md
[https://gfycat.com/](https://gfycat.com/)`name`
```

**Comment by cramforce** [comment]
Ping @googlebot 

Should this get in?

File: extensions/amp-gfycat/amp-gfycat.md (hunk 1 above)
Comment (rsimha) [reply extensions/amp-gfycat/amp-gfycat.md]: My original suggestion was based on the fact that this is meant to be an example, and not an actual clickable link. Either way, if the link checker is satisfied, I have no objections :)

**Comment by pbakaus** [comment]
@cramforce yes please.

**Comment by cramforce** [comment]
Merging per @pbakaus

**Comment by pbakaus** [comment]
@cramforce thanks much!

**Pull request merged**